remote-wallet/src/ledger.rs

```diff
@@ -498,7 +504,16 @@ impl RemoteWallet<hidapi::DeviceInfo> for LedgerWallet {
             chunks.last_mut().unwrap().0 &= !P2_MORE;
 
             for (p2, payload) in chunks {
-                result = self.send_apdu(commands::SIGN_MESSAGE, p1, p2, &payload)?;
+                result = self.send_apdu(
+                    if self.outdated_app() {
+                        commands::DEPRECATED_SIGN_MESSAGE
+                    } else {
+                        commands::SIGN_MESSAGE
+                    },
```

Comment:
yeah, I noticed that multi-chunk legacy stuff was broken for a while, large defi transactions would fail, but since no one complained, maybe not so many legacy users left, and we could drop the support of the outdated app entirely. I spoke with Trent about it, and we decided to keep it for now. Communication protocol needs a larger re-work, and then we can reconsider.